C24_WMDE_iPad_DE_01

Sperling-0 opened the pull request:
https://phabricator.wikimedia.org/T376593

Sperling-0 committed:
Prepare campaign for C24_WMDE_ipad_DE_01

Ticket: https://phabricator.wikimedia.org/T376593

Sperling-0 committed:
Prepare CTRL banner for C24_WMDE_iPad_DE_01

- The average donation should be changed from 22,25 € to 22,49 €
- Added space between 99% -> 99 %
- The third option below the two buttons "Ja, ich möchte jährlich spenden, aber einen anderen Betrag." is removed from the 2nd form page.
- The first slide shows the data and time
- It has the progress bar
- It has the campaign day sentence
- It has the visitors-vs-donors sentence
- It has the soft close
- It has an additional paragraph on the 2nd form page
- It has "I already donated" button in the soft close, respective theme css files have been modified
- Update clean-up-2025 document

Ticket: https://phabricator.wikimedia.org/T376593

Co-authored-by: Abban Dunne <abban.dunne@wikimedia.de>

Sperling-0 committed:
Implement VAR for C24_WMDE_iPad_DE_01

- The average donation should be changed from 22,25 € to 22,49 €
- Added space between 99% -> 99 %
- The third option below the two buttons "Ja, ich möchte jährlich spenden, aber einen anderen Betrag." is removed from the 2nd form page.
- The first slide shows the data and time
- The headline of the variant banner is changed
- The amount on the 3rd slide should also be changed from 5 € to 15 €
- The second amount option in the variant banner form should be 15 €
- It has the progress bar
- It has the campaign day sentence
- It has the visitors-vs-donors sentence
- It has "I already donated" button in the soft close
- Update content repo

Ticket: https://phabricator.wikimedia.org/T376593

Co-authored-by: Abban Dunne <abban.dunne@wikimedia.de>

Abban commented on banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:
```diff
@@ -5,12 +5,13 @@
 	$select-group-button: true,
```
When https://github.com/wmde/fundraising-banners/pull/579 is merged the payment logos here will break. You should be able to already add `$select-group-image-label: true` to this list.

Abban commented on banners/pad/C24_WMDE_iPad_DE_01/styles/styles.scss:
```diff
@@ -5,12 +5,13 @@
 @use 'MainBanner' with (
 	$banner-height: 378px,
-	$form-width: 256px
+	$form-width: 35%
```
Pixel width is used here is so the form remains consistent on all screen sizes. Is there a reason you changed it to a percentage?

Sperling-0 replied:
I did not change it. It came from `styles_var.scss` of `C24_WMDE_iPad_00`.

moiikana replied:
maybe we can try going back to the pixel value here

Abban commented on src/themes/Treedip/ProgressBar/ProgressBar.scss:
```diff
@@ -42,7 +42,7 @@ $progress-bar-margin: 0 !default;
 			position: relative;
 			background: var( --progress-bar-wrapper-background );
-			border: 3px var( --progress-bar-wrapper-border );
+			border: 3px var( --progress-bar-wrapper-border ) solid;
```
This works, but the general convention for the order when defining border styles is `border: <size> <style> <colour>;`

```css
border: 3px solid var( --progress-bar-wrapper-border );
```

Abban commented on banners/pad/C24_WMDE_iPad_DE_01/messages_var.ts:
This is identical to `messages.js` so you can delete this and use the same file in both banners.

Abban reviewed (approved):
It's looking good, just needs a few tweaks.

Sperling-0 commented:
⚠️ Rebase this PR from latest main.

To get the latest UoF working:

- For Desktop banners: `BannerCtrl.vue`, `BannerVar.vue`, `FallbackBanner.vue` needs to be updated. (If it is DE desktop include DE infographic and vice versa)
- For Mobile/ Ipad banners: `BannerCtrl.vue`, `BannerVar.vue` needs to be updated. (If it is DE mobile include DE infographic and vice versa)

Sperling-0 commented:
> ⚠️ Rebase this PR from latest main.
> 
> To get the latest UoF working:
> 
>     * For Desktop banners: `BannerCtrl.vue`, `BannerVar.vue`, `FallbackBanner.vue` needs to be updated. (If it is DE desktop include DE infographic and vice versa)
> 
>     * For Mobile/ Ipad banners: `BannerCtrl.vue`, `BannerVar.vue` needs to be updated. (If it is DE mobile include DE infographic and vice versa)

✅ Done

moiikana commented:
consider rebasing because https://github.com/wmde/fundraising-banners/pull/597 got merged

Sperling-0 commented:
> consider rebasing because #597 got merged

✅ Done

moiikana commented on banners/pad/C24_WMDE_iPad_DE_01/banner_var.ts:
```diff
@@ -0,0 +1,67 @@
+import { LocalImpressionCount } from '@src/utils/LocalImpressionCount';
+
+// Channel specific form setup
+import { createFormItems } from './form_items_var';
```
the ticket doesn't say anywhere that the variant should have different form items. is this a bug that was missed?

Sperling-0 replied:
![Screenshot from 2024-10-29 14-25-12](https://github.com/user-attachments/assets/5eef0b77-fea7-4979-bc6f-9e945b396536)

Please see the last point for the VAR banner.

That is what was talked about in the daily today. For ipad-01 and wpde-mob-01 banners the ticket kept renewing and new changes kept coming.

moiikana replied:
thanks! didn't see that.

Sperling-0 commented:
> [!Warning]
> Please do not merge this. The tests (timer and cookie image) are filing locally for me. So maybe this needs investigation.

Sperling-0 commented:
> Warning
> 
> Please do not merge this. The tests (timer and cookie image) are filing locally for me. So maybe this needs investigation.

> [!NOTE]  
> After the rebase to the latest main branch. All the tests are passing!!!

gbirke commented:
Could you move `tests/banners/pad/components` to `tests/banners/pad/C24_WMDE_iPad_DE_01/components` to prepare for #594 ? You can do it in a separate commit

Sperling-0 committed:
Move tests to the dedicated campaign folder

- https://github.com/wmde/fundraising-banners/pull/594

Ticket: https://phabricator.wikimedia.org/T369276

Sperling-0 commented:
> Could you move `tests/banners/pad/components` to `tests/banners/pad/C24_WMDE_iPad_DE_01/components` to prepare for #594 ? You can do it in a separate commit

✅ Done
